### DISCUSSION TITLE
Custom OAuth provider

### DISCUSSION THREAD

**Discussion by tigattack** [diced/zipline#325] [Ideas]
A custom OAuth provider to allow using self-hosted SSO would be great.

The way I imagine this being implemented is environment variables for all the necessary values, e.g. issuer URL, client ID, client secret, scopes, etc.

My use case is I'd like to use this with [Authentik](https://github.com/goauthentik/authentik).

**Comment by diced** [comment]
This has been suggested before, and I do feel like it would be great to add into Zipline. Unfortunately I'm not current working on any additional OAuth providers for Zipline, although it would be great if could contribute and make a pull request with the feature added.

**Comment by diced** [comment]
This is also a duplicate of https://github.com/diced/zipline/discussions/321
